terrain/voxel_terrain.cpp

```diff
@@ -384,7 +375,15 @@ void VoxelTerrain::start_updater() {
 
 	// TODO Thread-safe way to change those parameters
 	VoxelMeshUpdater::MeshingParams params;
-	params.smooth_surface = _smooth_meshing_enabled;
+
+	if (_stream.is_valid()) {
+		if (_stream->get_used_channels_mask() & 1<<VoxelBuffer::CHANNEL_TYPE) {
+			params.smooth_surface = false;
```

Comment:
A stream can provide both, but this condition will turn off smooth in that case.

Reply:
done, with the other two